### PR TITLE
Fix in-function storage class validation

### PR DESCRIPTION
**Pull request by dekimir**
Move tests into a fixture that properly turns on the relevant passes.

Test all storage classes.

### CODE REVIEW

**Comment by dekimir** [comment]
@umar456 PTAL

File: test/Validate.Storage.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+%vfunct = OpTypeFunction %voidt
+%ptrt   = OpTypePointer Function %intt
+%unicon = OpVariable %ptrt UniformConstant
+%input  = OpVariable %ptrt Input
+%unif   = OpVariable %ptrt Uniform
+%output = OpVariable %ptrt Output
+%wgroup = OpVariable %ptrt Workgroup
+%xwgrp  = OpVariable %ptrt CrossWorkgroup
+%priv   = OpVariable %ptrt Private
+%gener  = OpVariable %ptrt Generic
```
Comment (umar456) [inline test/Validate.Storage.cpp]: OpVariable storage class cannot be Generic

**Comment by dekimir** [comment]
Removed `Generic`.  Will report an error for it in the next PR.

**Comment by umar456** [comment]
Looks good to me.

**Comment by dneto0** [comment]
+2

**Comment by dekimir** [comment]
Merged in 8177fbe.

**Pull request closed without merging**